src/blob/core.rs

```diff
@@ -291,40 +301,50 @@ where
         key: &K,
         meta: Option<&Meta>,
         check_filters: bool,
-    ) -> Result<Vec<u8>> {
+    ) -> Result<ReadResult<Vec<u8>>> {
         debug!("blob read any");
-        let entry = self
-            .get_entry(key, meta, check_filters)
-            .await?
-            .ok_or_else(|| Error::from(ErrorKind::RecordNotFound))?;
-        debug!("blob read any entry found");
-        let buf = entry
-            .load()
-            .await
-            .with_context(|| {
-                format!(
-                    "failed to read key {:?} with meta {:?} from blob {:?}",
-                    key,
-                    meta,
-                    self.name.to_path()
-                )
-            })?
-            .into_data();
-        debug!("blob read any entry loaded bytes: {}", buf.len());
-        Ok(buf)
+        let entry = self.get_entry(key, meta, check_filters).await?;
+        match entry {
+            ReadResult::Found(entry) => {
+                debug!("blob read any entry found");
+                let buf = entry
+                    .load()
+                    .await
+                    .with_context(|| {
+                        format!(
+                            "failed to read key {:?} with meta {:?} from blob {:?}",
+                            key,
+                            meta,
+                            self.name.to_path()
+                        )
+                    })?
+                    .into_data();
+                debug!("blob read any entry loaded bytes: {}", buf.len());
+                Ok(ReadResult::Found(buf))
+            }
+            ReadResult::Deleted(ts) => Ok(ReadResult::Deleted(ts)),
+            ReadResult::NotFound => Ok(ReadResult::NotFound),
+        }
     }
 
     #[inline]
-    pub(crate) async fn read_all_entries(&self, key: &K) -> Result<Option<Vec<Entry>>> {
+    pub(crate) async fn read_all_entries(&self, key: &K) -> Result<ReadResult<Vec<Entry>>> {
         let headers = self.index.get_all(key).await?;
-        Ok(headers.map(|h| {
-            debug!("blob core read all {} headers", h.len());
-            Self::headers_to_entries(h, &self.file)
+        Ok(headers.map(|hs| {
+            debug_assert!(hs
+                .iter()
+                .zip(hs.iter().skip(1))
+                .all(|(x, y)| x.created() <= y.created()));
```

Comment:
Looks like the condition should be reversed (`>=`)

Reply:
Latest should be the last, isn't it?...

Reply:
According to the #209 it is the first. I think this is a better order (most recent entries first)